web/js/codeworld-mode.js

```diff
@@ -82,43 +116,91 @@ CodeMirror.defineMode('codeworld', (_config, modeConfig) => {
 
         if (stream.match(RE_DASHES)) {
             stream.skipToEnd();
-            return 'comment';
+            return ['comment'];
         }
 
-        if (stream.match(RE_QUAL)) return 'qualifier';
-        if (stream.match(RE_VARID) || stream.match(RE_VARSYM)) return 'variable';
-        if (stream.match(RE_CONID) || stream.match(RE_CONSYM)) return 'variable-2';
-        if (stream.match(RE_NUMBER)) return 'number';
-        if (stream.match(RE_CHAR) || stream.match(RE_STRING)) return 'string';
+        if (checkContextStarter(stream, state)) {
+            return ['keyword'];
+        }
+
+        if (state.scanState === NORMAL && stream.match(RE_IN)) {
+            // Cancel previous layout 'let' context
+            state.layoutContext.pop();
+            return ['keyword'];
+        } else if (state.scanState === NORMAL) {
+            let lastContext = state.layoutContext[state.layoutContext.length - 1];
+            lastContext = lastContext === '{' ? 0 : lastContext;
+            if (stream.sol() && stream.indentation() < lastContext) {
+                // Close previous context
```

Comment:
I think you need to close layout contexts even if you're in TRIGGERED state.  For instance:

    foo = bar
      where a = case 42 of

    baz = 43

Here, `baz` should still break the layout context for `foo`.

Reply:
TRIGGERED does mark next token as open of new context, wherever it is.
![image](https://user-images.githubusercontent.com/5370736/54473154-0c3e4080-47e5-11e9-8b0e-fd39e3a8bd44.png)

For your example there should be condition like "if current column less than `where` indentation - do not open new context". Or even close all contexts, which context starter indentation more than current column. I see next solution - make context entries `(contextStarterColumn, contextColumn)` instead of just `contextColumn` integer value.